### PR TITLE
Refactored in memory event store

### PR DESCRIPTION
**Pull request by thoozaho**
Both event store implementations now use the same in-memory data
structure.

Tests that are implementation agnostic have been moved to
EventStoreTests and are tested against both. Kafka specific
tests have their own class.

`Offset` (which is a Kafka specific term) has been renamed `Version`

### CODE REVIEW

**Comment by thoozaho** [comment]
@og3b - this came about as I needed to add a `surface` method to the EventStore and would have needed to implement it twice.

Both EventStore implementations use the same in-memory data structure now - their difference is in their `write` methods.

There are no new tests - they have just been moved around. Implementation agnostic tests are now done against both implementations.

File: eventstore/src/evcel/eventstore/EventStore.scala [hunk 1]
```diff
@@ -1,442 +1,93 @@
...
-            } catch {
-              case e : Exception =>
-                log.error(s"Error processing queue $name, shutting down", e)
-                shutdown()
-            }
-          }
+  def surface(offset : Version) : Map[K, V] = {
+    if (offset > lastVersionInHistory())
+      throw new RuntimeException(
+        s"$name asked for surface for offset $offset when latest is $lastVersionInHistory()")
```
Comment (og3b) [inline eventstore/src/evcel/eventstore/EventStore.scala]: the class is now called Version but the instances and messages still refer to offset.

Comment (thoozaho) [inline eventstore/src/evcel/eventstore/EventStore.scala]: Oh yes - I'll fix that

File: eventstore/tests/evcel/eventstore/EventStoreTests.scala [hunk 2]
```diff
@@ -1,144 +1,70 @@
...
-      (uuid, map).toJson.prettyPrint
+/*
+ * These are tests that aren't specific to any particular
+ * implementation of EventStore. 
+ */
+class EventStoreTests extends FreeSpec with Matchers{
+
+  private def withEventStore(fn: EventStore[String, String] => Unit) {
+    def executeWithKafkaStore(){
+      KafkaTestUtils.withTestKafka {
```
Comment (og3b) [inline eventstore/tests/evcel/eventstore/EventStoreTests.scala]: These tests test the kafka eventstore and so do the kafka eventstore tests. Should this test just test the in memory one? Not sure why the split if this one is testing kafka too.

Comment (thoozaho) [inline eventstore/tests/evcel/eventstore/EventStoreTests.scala]: These tests test all implementations of EventStore - KafkaEventStoreTests is to test things that are specific to the Kafka implementation.

**Comment by og3b** [comment]
@thoozaho looks good. merge when ready.

**Comment by thoozaho** [comment]
@og3b - crap - forgot to push the s/offset/version change first - I'll do that now.

**Pull request merged**